sdk/aqueduct/artifacts/utils.py

```diff
@@ -35,6 +36,20 @@ def preview_artifacts(
     Returns a list of artifacts, each corresponding to one of the provided `target_artifact_ids`, in
     the same order.
     """
+    if globals.__GLOBAL_CONFIG__.engine is not None:
+        engine = globals.__GLOBAL_CONFIG__.engine
+        if engine is None:
+            engine_config = EngineConfig()
+        else:
+            connected_integrations = globals.__GLOBAL_API_CLIENT__.list_integrations()
+            if engine not in connected_integrations.keys():
+                raise InvalidIntegrationException(
+                    "Not connected to compute integration %s!" % engine
+                )
+            engine_config = generate_engine_config(connected_integrations[engine])
+        dag.engine_config = engine_config
+        assert dag.engine_config is not None
```

Comment:
Chatted with @Fanjia-Yan:
- The `set_engine_config()` comment still holds.
- It turns out that leaving the `engine_config` field on the DAG (SDK side) empty doesn't matter, since its backfilled to the default engine config on the backend: https://github.com/aqueducthq/aqueduct/blob/main/src/golang/cmd/server/request/dag.go#L76. @Fanjia-Yan could you add a comment on the field in `dag.py` that a None value means default aqueduct server?